src/QuadForm/Torsion.jl

```diff
@@ -733,17 +759,383 @@ function preimage(f::TorQuadModMor, a::TorQuadModElem)
+# "Not yet implemented". If yes, we compare the normal forms of the respective complements
+# which are non-degenerate, and if the radicals have the same elementary divisors, we
+# complete the isometry by adding the identity matrix from one radical to the other one.
+function _isometry_degenerate(T::TorQuadMod, U::TorQuadMod)
+  # the zero map for default output
+  hz = hom(T, U, zero_matrix(ZZ, ngens(T), ngens(U)))
+  rqT, rqTtoT = radical_quadratic(T)
+  rqU, rqUtoU = radical_quadratic(U)
+  if elementary_divisors(rqT) != elementary_divisors(rqU)
+    return (false, hz)
+  end
+  # at this point we can map safely one radical to the other one
+  boolT, jT = has_complement(rqTtoT)
+  boolU, jU = has_complement(rqUtoU)
+  @req boolT && boolU "Not yet implemented" # Non split case
+  NT = domain(jT)
+  NU = domain(jU)
+  bool, isom = _isometry_semiregular(NT, NU)
+  if !bool
+    return (false, hz)
+  end
+  # now we know that there is an isometry, just need to put everything together
+  # we first tidy the generators of the radicals up
+  geneT = TorQuadModElem[]
+  for a in gens(rqT)
+    if (!(a in geneT) && !iszero(a))
+      push!(geneT, a)
+    end
+  end
+  @assert sort(order.(geneT)) == sort(elementary_divisors(rqT))
+  geneU = TorQuadModElem[]
+  for a in gens(rqU)
+    if (!(a in geneU) && !iszero(a))
+      push!(geneU, a)
+    end
+  end
+  @assert sort(order.(geneU)) == sort(elementary_divisors(rqU))
+  # we map generators of the radical and its complement in the module
+  # to obtain an isomorphic module with a nicer basis
+  geneT = rqTtoT.(geneT)
+  append!(geneT, jT.(gens(NT)))
+  Tsub, TsubinT = sub(T, geneT)
+  @assert is_bijective(TsubinT)  # same module, different bases, since we have a splitting
+  geneU = rqUtoU.(geneU)
+  append!(geneU, jU.(gens(NU)))
+  Usub, UsubinU = sub(U, geneU)
+  @assert is_bijective(UsubinU)
+  @assert length(geneT) == length(geneU)
+  # now the radical parts are similar, the normal parts are isometric, we just
+  # need to create our bijective mapping by sending generators of one radical to the
+  # other and applying our previously computed isometry to the complements
+  I = identity_matrix(ZZ, length(geneT)-length(gens(NT))) # for the radicals
+  M = isom.map_ab.map                                     # for the complements
+  D = block_diagonal_matrix([I, M])
+  phi = hom(Tsub, Usub, D)
+  @assert is_bijective(phi)
+  TtoU = compose(inv(TsubinT), compose(phi, UsubinU))
+  @assert all(a -> a*a == TtoU(a)*TtoU(a), gens(T))
+  return (true, TtoU)
+end
+
+@doc Markdown.doc"""
+    is_isometric_with_map(T::TorQuadMod, U::TorQuadMod)
+                                                   -> Bool, TorQuadModMor
+
+Return whether the torsion quadratic modules `T` and `U` are isometric.
+If yes, it also returns an isometry $T \to U$.
+
+If `T` and `U` are degenerate it requires that they both split into a direct
+sum of their respective quadratic radical (see `radical_quadratic`).
```

Comment:
```suggestion
sum of their respective quadratic radical (see [`radical_quadratic`](@ref).
```
To make this reference a link in the manual.